### PR TITLE
Run 5188/part 2

### PR DESCRIPTION
**Pull request by pbaize**
#### Description of Change
Follow up to part 1. Routes webcontents related apis separately so they can be shared with browserview

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] PR title starts with the JIRA ticket [pull request process](https://github.com/openfin/Internal-Wiki/wiki/Pull-Request-Process)

#### Release Notes

Notes: n/a

### CODE REVIEW

File: src/browser/api/webcontents.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import * as url from 'url';
+
+export function executeJavascript(webContents: Electron.WebContents, code: string, callback: (e: any, result: any) => void): void {
```
Comment (tomer-openfin) [inline src/browser/api/webcontents.ts]: Maybe we want to use the opportunity to move to a Promise flow here?

File: src/browser/api/webcontents.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        url: webContents.getURL()
+    };
+}
+
+export function getAbsolutePath(webContents: Electron.WebContents, path: string) {
+    const windowURL = webContents.getURL();
+    return  url.resolve(windowURL, path);
+}
+
+export function navigate (webContents: Electron.WebContents, url: string) {
```
Comment (MichaelMCoates) [inline src/browser/api/webcontents.ts]: minor detail, but spacing consistency after function names. Feel free to ignore.

File: src/browser/api/window.js [hunk 3]
```diff
@@ -35,6 +34,7 @@ import { toSafeInt } from '../../common/safe_int';
 import route from '../../common/route';
 import { FrameInfo } from './frame';
 import { System } from './system';
+import * as WebContents from './webcontents';
```
Comment (tomer-openfin) [inline src/browser/api/window.js]: an opinionated note -imo it's nicer when there's 1 default export containing an object with all the methods in it, along with individual exports for the methods. That way this line turns to 

> import WebContents from './webcontents';

File: src/browser/api/window.js [hunk 4]
```diff
@@ -1272,23 +1270,16 @@ Window.getGroup = function(identity) {
...
     return windowInfo;
 };
 
 
 Window.getAbsolutePath = function(identity, path) {
     let browserWindow = getElectronBrowserWindow(identity, 'get URL for');
-    let windowURL = browserWindow.webContents.getURL();
-
-    return (path || path === 0) ? url.resolve(windowURL, path) : '';
+    return (path || path === 0) ? WebContents.getAbsolutePath(browserWindow.webContents, path) : '';
```
Comment (tomer-openfin) [inline src/browser/api/window.js]: why would path equal 0?

File: src/browser/api_protocol/api_handlers/webcontents.ts [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+            });
+        }
+        pUuid = Application.getParentApplication({
+            uuid: pUuid
+        });
+    }
+
+    return nack(new Error('Rejected, target window is not owned by requesting identity'));
+}
+function navigateWindow(identity: Identity, message: APIMessage, ack: Acker, nack: (error: Error) => void): void {
```
Comment (tomer-openfin) [inline src/browser/api_protocol/api_handlers/webcontents.ts]: Does the name still make sense after moving out of window?

Comment (tomer-openfin) [reply src/browser/api_protocol/api_handlers/webcontents.ts]: (this applies to other methods here too)

Comment (pbaize) [reply src/browser/api_protocol/api_handlers/webcontents.ts]: I'd rather rename in a follow up pr, for now it still only does windows and I'm trying to keep changes to a minimum

Comment (rdepena) [reply src/browser/api_protocol/api_handlers/webcontents.ts]: No rename is needed, the API action cannot change `navigate-window` and the only purpose of the API name is to be mapped.

Comment (pbaize) [reply src/browser/api_protocol/api_handlers/webcontents.ts]: @rdepena I assume @tomer-openfin was referencing the function names, not the api mapping

**Comment by rdepena** [comment]
👍

File: src/browser/api_protocol/api_handlers/webcontents.ts (hunk 5 above)
Comment (rdepena) [reply src/browser/api_protocol/api_handlers/webcontents.ts]: right, but the function is only used by the mapping so it should just keep the name.

**Pull request merged**